File: app/src/main/java/me/tysheng/xishi/ui/EmailDialog.java
package me.tysheng.xishi.ui;

import android.app.Dialog;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomSheetBehavior;
import android.support.design.widget.BottomSheetDialog;
import android.support.design.widget.BottomSheetDialogFragment;
import android.view.View;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;

import java.util.ArrayList;

import me.tysheng.xishi.R;
import me.tysheng.xishi.adapter.BaseLoadMoreRecyclerViewAdapter;
import me.tysheng.xishi.databinding.DialogBottomSheetBinding;
import me.tysheng.xishi.utils.SystemUtil;

/**
 * Created by Sty
 * Date: 16/8/18 22:26.
 */
public class EmailDialog extends BottomSheetDialogFragment {
    private BottomSheetBehavior mBehavior;

    @Override
    public void onStart() {
        super.onStart();
        //默认全屏展开
        mBehavior.setState(BottomSheetBehavior.STATE_EXPANDED);
//        getDialog().getWindow().getDecorView().setBackgroundColor(Color.TRANSPARENT);
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        BottomSheetDialog dialog = (BottomSheetDialog) super.onCreateDialog(savedInstanceState);
        DialogBottomSheetBinding binding = DataBindingUtil.inflate(getActivity().getLayoutInflater(), R.layout.dialog_bottom_sheet, null, false);
        ArrayList<Bean> list = new ArrayList<>();
        list.add(new Bean(R.drawable.ic_email_black_24dp, "邮件反馈"));
        list.add(new Bean(R.drawable.ic_update_black_24dp, "检查更新"));
        list.add(new Bean(R.drawable.ic_thumb_up_black_24dp, "捐赠支持"));
        list.add(new Bean(R.drawable.ic_content_copy_black_24dp, "复制邮箱"));
        list.add(new Bean(R.drawable.ic_compare_arrows_black_24dp, "主题切换"));
        binding.title.setText(String.format("版本 %s", SystemUtil.getVersionName()));
        binding.shareRecyclerView.setHasFixedSize(true);
        BaseLoadMoreRecyclerViewAdapter<Bean> adapter = new BaseLoadMoreRecyclerViewAdapter<Bean>(R.layout.item_dialog_bottom_sheet, list) {
            @Override
            protected void convert(BaseViewHolder holder, Bean item) {
                holder.setText(R.id.title, item.name)
                        .setImageResource(R.id.icon, item.res);
            }
        };
        adapter.setOnItemClickListener(new BaseQuickAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(BaseQuickAdapter adapter, View view, int position) {
                ((MainActivity) getContext()).onItemClick(position);
                dismiss();
            }
        });
        adapter.bindToRecyclerView(binding.shareRecyclerView);
        dialog.setContentView(binding.getRoot());
        mBehavior = BottomSheetBehavior.from((View) binding.getRoot().getParent());
        return dialog;
    }

    private static class Bean {
        int res;
        String name;

        Bean(int res, String name) {
            this.res = res;
            this.name = name;
        }
    }
}

File: app/src/main/java/me/tysheng/xishi/ui/EmailDialog.kt
package me.tysheng.xishi.ui

import android.app.Activity
import android.app.Dialog
import android.os.Bundle
import android.support.design.widget.BottomSheetBehavior
import android.support.design.widget.BottomSheetDialog
import android.support.design.widget.BottomSheetDialogFragment
import android.view.View
import com.chad.library.adapter.base.BaseViewHolder
import kotlinx.android.synthetic.main.dialog_bottom_sheet.view.*
import me.tysheng.xishi.BuildConfig
import me.tysheng.xishi.R
import me.tysheng.xishi.adapter.BaseLoadMoreRecyclerViewAdapter

/**
 * Created by Sty
 * Date: 16/8/18 22:26.
 */
class EmailDialog : BottomSheetDialogFragment() {
    private var behavior: BottomSheetBehavior<*>? = null
    var dialogCallback: DialogCallback? = null
    private lateinit var customView: View
    override fun onStart() {
        super.onStart()
        behavior?.state = BottomSheetBehavior.STATE_EXPANDED
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dialog = super.onCreateDialog(savedInstanceState) as BottomSheetDialog
        customView = requireActivity().layoutInflater.inflate(R.layout.dialog_bottom_sheet, null)
        dialog.setContentView(customView)
        behavior = BottomSheetBehavior.from(customView.parent as View)
        return dialog
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        val list = arrayListOf(R.drawable.ic_email_black_24dp to getString(R.string.feedback_by_email),
                R.drawable.ic_update_black_24dp to getString(R.string.check_update),
                R.drawable.ic_thumb_up_black_24dp to getString(R.string.donate),
                R.drawable.ic_content_copy_black_24dp to getString(R.string.copy_email),
                R.drawable.ic_compare_arrows_black_24dp to getString(R.string.change_theme))
        customView.title.text = ("${getString(R.string.version)} ${BuildConfig.VERSION_NAME}")
        object : BaseLoadMoreRecyclerViewAdapter<Pair<Int, String>>(R.layout.item_dialog_bottom_sheet, list) {
            override fun convert(holder: BaseViewHolder, item: Pair<Int, String>) {
                holder.setText(R.id.title, item.second)
                        .setImageResource(R.id.icon, item.first)
            }
        }.apply {
            bindToRecyclerView(customView.shareRecyclerView)
            setOnItemClickListener { _, _, position ->
                dismissAllowingStateLoss()

                dialogCallback?.itemClick(position)
            }
        }
    }

    companion object {
        const val TAG = "EmailDialog"
    }
}

interface DialogCallback {
    fun itemClick(position: Int)
}

sealed class MainDialogActionListener
class SendEmail(val activity: Activity) : MainDialogActionListener()
class ShareToStore(val activity: Activity) : MainDialogActionListener()
class JumpToAlipay(val activity: Activity) : MainDialogActionListener()
class CopyEmail(val activity: Activity) : MainDialogActionListener()
class SwitchDayNightMode(val activity: Activity) : MainDialogActionListener()